BugFix segfault caused due to missing name param

AZaugg opened the pull request:
If a repository definition is missing the name parameter the repositroy is later processed and a segfault occurs.

Adding a check to ensure that the repository is not processed in the event that the name parameter is missing.

fixes #400

oliverkurth commented on client/init.c:
```diff
@@ -251,8 +251,9 @@ TDNFRefreshSack(
+         * In addition skip any repository that has a null repo name set
          */
-        if (!strcmp(pRepo->pszName, CMDLINE_REPO_NAME) || !pRepo->nEnabled)
+        if ((!pRepo->pszName) || !strcmp(pRepo->pszName, CMDLINE_REPO_NAME) || !pRepo->nEnabled)
```
I think we should not just ignore and skip it. I think we should check for it in line 238, and bail if NULL.

Maybe we should also add a check right when we read the configuration.

oliverkurth replied:
Can you please address this?

oliverkurth replied:
@AZaugg , are going to address my comment? It's fine if not, I can take over.

AZaugg committed:
BugFix segfault caused due to missing name param

If a repository definition is missing the name parameter
the repositroy is later processed and a segfault occurs.

Set the repository name to the repo stanza in the event that
the name parameter is missing.

fixes #400

AZaugg commented:
OK, I changed my approach, as the repo stanzas need to be unique, I set the name value to the stanza definition as the default value. I think this is the best of both worlds. Not ignoring it, but not failing if a repo file is missing the name param

oliverkurth commented:
> OK, I changed my approach, as the repo stanzas need to be unique, I set the name value to the stanza definition as the default value.

That makes a lot of sense.
